src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -283,16 +283,403 @@ std::vector<uint8_t> run_and_save_to_memory(std::istringstream & test_idl, std::
+        VerifyReturnType(returnType, returnTypeRef, invokeMethod, typeDef);
+
+        auto param(params.begin());
+        for (size_t i = 1; i < size(invokeMethod.ParamList()); i++)
+        {
+            param->VerifyType(invokeMethod.ParamList().first[i], invokeMethod.Signature().Params().first[i - 1], typeDef.TypeNamespace());
+            ++param;
+        }
+    }
+};
+
+struct ExpectedMethod : ExpectedType {
+    ElementType returnType;
+    std::string returnTypeRef;
+    std::vector<ExpectedParam> params;
+
+    ExpectedMethod(const std::string &expectedName, const ElementType expectedReturnType, std::vector<ExpectedParam> &&expectedParams)
+        : ExpectedType(expectedName), returnType(expectedReturnType), returnTypeRef(""), params(expectedParams) {}
+    ExpectedMethod(const std::string &expectedName, const std::string &expectedReturnType, std::vector<ExpectedParam> &&expectedParams)
+        : ExpectedType(expectedName), returnType(ElementType::End), returnTypeRef(expectedReturnType), params(expectedParams) {}
+
+    void VerifyType(MethodDef const& method) const override
+    {
+        REQUIRE(method.Name() == name);
+
+        VerifyReturnType(returnType, returnTypeRef, method, method.Parent());
+
+        auto param(params.begin());
+        for (size_t i = 1; i < size(method.ParamList()); i++)
+        {
+            param->VerifyType(method.ParamList().first[i], method.Signature().Params().first[i - 1], method.Parent().TypeNamespace());
+            ++param;
+        }
+    }
+};
+
+struct ExpectedProperty : ExpectedType {
+    ElementType type;
+    std::string typeRef;
+
+    ExpectedProperty(const std::string &expectedName, const ElementType expectedType)
+        : ExpectedType(expectedName), type(expectedType), typeRef("") {}
+    ExpectedProperty(const std::string &expectedName, const std::string &expectedType)
+        : ExpectedType(expectedName), type(ElementType::End), typeRef(expectedType) {}
+
+    void VerifyType(Property const& prop) const override
+    {
+        REQUIRE(prop.Name() == name);
+        REQUIRE(prop.Flags().value == method_attributes_no_flags().value);
+
+        if (type == ElementType::End)
+        {
+            VerifyTypeRefName(typeRef, prop.Parent().TypeNamespace(), std::get<coded_index<TypeDefOrRef>>(prop.Type().Type().Type()).TypeRef());
+        }
+        else
+        {
+            REQUIRE(prop.Type().Type().element_type() == type);
+        }
+
+        bool foundGetMethod = false;
+        bool foundSetMethod = false;
+        std::string getMethodName = "get_";
+        std::string setMethodName = "put_";
+        getMethodName += prop.Name();
+        setMethodName += prop.Name();
+        for (auto &&method : prop.Parent().MethodList())
+        {
+            if (method.Flags().SpecialName() == true)
+            {
+                if (getMethodName == method.Name())
+                {
+                    foundGetMethod = true;
```

Comment:
We should check the signature of the return type here for the get method. It should be the type of the property.